File: top scoring student.py
"""
INPUT:
4
Sasikumar:50:60:70
Arun:60:40:90
Manoj:50:50:60
Rekha:60:35:45
OUTPUT:
Arun
"""


n=int(input())
max=0
for i in range(n):
    inp=input().split()
    for j in range(len(inp)):
        name,m,p,c=inp[j].split(':')
        m,p,c=int(m),int(p),int(c)
        sum=m+p+c
        if sum>max:
            max=sum
            maxstud=name
print(maxstud)

File: top scoring student.java
"""
INPUT:
4
Sasikumar:50:60:70
Arun:60:40:90
Manoj:50:50:60
Rekha:60:35:45
OUTPUT:
Arun
"""
import java.util.*;
public class Hello
{
public static void main(String args[])
{
Scanner in=new Scanner(System.in);
int n=in.nextInt();
int maxavg=0;
String top=" ";
for(int i=0;i<n;i++)
{
  String str=in.next();
  String stud=" ";
  int avg,sum=0;
  String[] st=str.split(":");
  for(int j=0;j<st.length;j++)
  {
    int run=0;
    if(j==0)
    {
      stud=st[j];
    }
    else
    {
      run=Integer.parseInt(st[j]);
      sum+=run;
    }
  }
  avg=sum/3;
  if(avg>maxavg)
  {
    maxavg=avg;
    top=stud;
  }
}

System.out.print(top);
}
}
